client_plugin/utils/refcount/refcnt.go

```diff
@@ -428,6 +430,16 @@ func (r *RefCountsMap) syncMountsWithRefCounters(d drivers.VolumeDriver) {
 	defer r.mtx.Unlock()
 
 	for vol, cnt := range r.refMap {
+		// If cnt is nil then this volume entry under mount root can be
+		// purged and the volume detached (if it is attached).
+		if cnt == nil {
```

Comment:
See my comment below. Rather than depending on NULL entry, we should switch-case based on mounted | refCnt | something else?